service/dap/server.go

```diff
@@ -1859,6 +1876,53 @@ func (s *Server) childrenToDAPVariables(v *fullyQualifiedVariable) ([]dap.Variab
 				Value:              cvalue,
 				VariablesReference: cvarref,
 				IndexedVariables:   getIndexedVariableCount(c),
+				NamedVariables:     getNamedVariableCount(c),
+			}
+		}
+	}
+	return children, nil
+}
+
+func getNamedVariableCount(v *proc.Variable) int {
+	namedVars := 0
+	if len(v.Children) > 0 && (v.Kind == reflect.Array || v.Kind == reflect.Slice) {
+		childKind := v.Children[0].RealType.Common().ReflectKind
+		if len(v.Children) > 0 && (childKind == reflect.Uint8 || childKind == reflect.Int32) {
+			// string value of array/slice of bytes and runes.
+			namedVars += 1
+		}
+	}
+	return namedVars
+}
+
+// toNamedChildren returns the DAP presentation of the referenced variable's named children.
+// Named variables for this server are used to send additional information about a variable
+// that is separate from the indexed variables.
```

Comment:
Done.

Though I do wonder if we should make all children be able to be paged if necessary (struct with 100s of fields shouldn't appear, but if it did we might want to page the results.

Reply:
The dap spec talks about named variables as a way to page them. I was thinking that it could be useful for everything else we returned currently without a filter.. But I was unable to get it to work. Have you? I tried with a struct with more >100 fields. I also tried with a long array with 500 named and 500 indexed. The editor sent a request with "named" filter at the point where the variable got expanded, while the "indexed" filter kicks in when you expand one of the [m..n] pages.

Reply:
Just wanted to add. You had to use named variables vs indexed to keep this field separate from all the elements that would fall under paging. But what if you want to include a metadata field into a struct or some other non-indexed type? In that case, won't you have to make all of the fields named? Since the total number of children is not specified, the editor will have no way of telling that there will bee some default filter-less children if you set namedVariables.

Reply:
In VSCode they do not yet page named children, though there is an issue for it that is currently marked out of scope (https://github.com/microsoft/vscode/issues/13878). This also includes a use case for PHP of needing both named and indexed to show the children of a particular type.

I can't think of a scenario in Go where we would need both indexed and named children to represent a variable. Is there such a case? I think we could just make named the metadata children and indexed everything else (so childrenToDAPVariables would be the indexed children). We already use indexed for maps, which are not actually indexed so this would not be too much of a jump.

For these named children, this is only being applied to arrays and slices, which already implement the indexed children, so this should not affect any variable types that currently return 0 for indexed.

Reply:
Yes, yes, I was thinking the same thing. Obviously out of scope for the PR, but a good trigger for the discussion. I do worry that there isn't an easy way to map field name to index. But then again if you really need a specific field, how likely are you to scroll through hundreds by hand instead of using watch/repl. We could also display some fields as named and not paged by default and then page the rest.